### DISCUSSION TITLE
How to find outer edges?

### DISCUSSION THREAD

**Discussion by chaohuang** [libigl/libigl#1917] [Q&A]
If I understand correctly, an outer edge is an edge that is reachable from infinity without crossing any faces.

Let's consider a planar connected component of triangles, which is not simply connected. In other words, there are "holes" in the connected component. So, the boundary edges that forms the boundaries of the "holes" are NOT outer edges, and other edges are outer edges.

So given a planar connected component of triangles and one of its boundary edges, how to determine if this boundary edge is an outer edge or not? Or better, how to find all outer edges?

Thanks!

**Comment by alecjacobson** [comment]
I guess there are a number of ways to do this. For a planar connected component, perhaps the simplest would be to extract all boundary edges, separate those edges into connected loops, find the loop with the largest enclosed area.
